bin/remote-prover/src/api/prover.rs

```diff
@@ -277,8 +277,9 @@ mod test {
         let mut client_2 = ApiClient::connect("http://127.0.0.1:50052").await.unwrap();
 
         // Create a mock transaction to send to the server
-        let mut mock_chain = MockChain::new();
-        let account = mock_chain.add_pending_existing_wallet(Auth::BasicAuth, vec![]);
+        let mut mock_chain = MockChain::builder();
+        let account = mock_chain.add_existing_wallet_with_assets(Auth::BasicAuth, vec![]).unwrap();
+        let mut mock_chain = mock_chain.build().unwrap();
```

Comment:
@PhilippGackstatter @TomasArrachea any idea whether this diff, or anything else coming out of next miden-base could cause this error?

https://github.com/0xMiden/miden-node/actions/runs/16405132235/job/46349950672?pr=1094

```
thread 'api::prover::test::test_prove_transaction' panicked at bin/remote-prover/src/api/prover.rs:312:57:
called `Result::unwrap()` on an `Err` value: 
TransactionProgramExecutionFailed(FailedAssertion { 
label: SourceSpan { source_id: SourceId(4294967295), start: ByteIndex(0), end: ByteIndex(0) }, source_file: None, clk: RowIndex(63360), err_code: 6550476539395293665, err_msg: Some("account nonce can only be incremented once")
})
```

Reply:
This has been recently merged into `next` in `miden-base`. Basically, previously account nonce could have been incremented multiple times in the same transaction. Now, we allow incrementing nonce only once. Maybe some tests was incrementing the nonce more than once?

Reply:
Btw. the `note_1` that gets passed to `extend_input_notes` is completely ignored and is not part of the transaction. So the test was always executing with 0 input notes.

There is 1) a `mock_chain.prove_next_block()` missing, to add this note to the chain state and 2) the note ID should be passed directly to `build_tx_context`, otherwise it is ignored.

A better way to do 1) would be to add this note directly using the `MockChainBuilder` further above, so it is directly part of the genesis state.

As for the concurrency issue, I don't really understand why two proving requests are made in the first place and why one of them would be expected to fail and the other succeed.

Reply:
Thanks I've created an issue to follow up on the test after this PR https://github.com/0xMiden/miden-node/issues/1102